rust/parser/mod.rs

```diff
@@ -483,19 +485,20 @@ fn visit_variable_type(tree: SyntaxTree) -> TypeVariable {
             match keyword.as_rule() {
                 Rule::ABSTRACT => var_type.abstract_(),
                 Rule::OWNS => {
-                    let type_ = visit_type(constraint.consume_expected(Rule::type_));
-                    let overridden = constraint
-                        .consume_if_matches(Rule::AS)
-                        .and_then(|_| Some(visit_type(constraint.consume_expected(Rule::type_))));
-                    let is_key =
-                        IsKeyAttribute::from(constraint.consume_if_matches(Rule::IS_KEY).is_some());
-                    var_type.constrain_owns(OwnsConstraint::from((type_, overridden, is_key)))
+                    let type_ =
+                        visit_type(constraint.consume_expected(Rule::type_)).into_type_variable();
```

Comment:
this line looks too short / awkwardly split up. I think it would look  better like:
```
let type = visit_type(...)
    .into_type_variable()
```